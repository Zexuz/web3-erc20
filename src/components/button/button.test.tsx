import renderer from "react-test-renderer";
import { Button } from "./index.tsx";

describe("Button", () => {
  it("should render", () => {
    const tree = renderer
      .create(<Button text={"Test"} onClick={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
