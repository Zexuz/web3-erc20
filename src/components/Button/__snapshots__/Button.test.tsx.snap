// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="relative flex w-full items-center justify-center rounded-md px-4 py-2 text-white 
                 bg-primary"
  onClick={[Function]}
>
  <p
    className="font-poppins text-sm font-medium"
  >
    Test
  </p>
</button>
`;

exports[`Button should render with loading icon 1`] = `
<button
  className="relative flex w-full items-center justify-center rounded-md px-4 py-2 text-white 
                 bg-primary"
  onClick={[Function]}
>
  <p
    className="font-poppins text-sm font-medium"
  >
    Action
  </p>
  <div
    className="absolute right-4 loader"
  />
</button>
`;
